README.md

```diff
@@ -1,16 +1,83 @@
 # sureql
-Create JavaScript wrappers for PostgreSQL statements with named parameters
 
-> sureql is a fork of [yesql](https://github.com/pihvi/yesql) by [@pihvi](https://github.com/pihvi),
-> though the workflow has been dramatically changed (and is for now somewhat undocumented...)
+Create TypeScript wrappers for PostgreSQL statements with named parameters.
 
-# TODO
+## Examples
 
-- document the differences from upstream and the new CLI-based workflow
-- re-add tests from upstream?
+The CLI (exposed through the usual `npx sureql` / `yarn run sureql` workflows) expects 1+
+directories of `.sql` files [following the same rules and syntax as
+yesql](https://github.com/pihvi/yesql/blob/8327f3faa0458f547ab22d3db7b62e208355b645/README.md) and a
+TypeScript output directory as arguments, and will compile, for example, this:
 
-# Legal
+```sql
+-- selectUsingTwoDisparateKeys
+SELECT mt.blah_id
+FROM   public.mytable mt
+WHERE  mt.customer_id = :'customerId'
+AND    mt.date_of_purchase < :'dateOfPurchase'
+;
+```
 
-sureql is released under the ISC License, [the same terms as
-yesql](https://github.com/pihvi/yesql/blob/8327f3faa0458f547ab22d3db7b62e208355b645/package.json#L29).
+... into this, a TypeScript wrapper function which takes named parameters validated at compile
+time!:
+
+```typescript
+import { QueryConfig as PgQuery } from 'pg';
+
+export const argumentPattern = /(?<prefix>::?)(?<quote>['"]?)(?<key>[a-zA-Z0-9_]+)\k<quote>/g;
+export const rawQuery = `-- selectUsingTwoDisparateKeys
+SELECT mt.blah_id
+FROM   public.mytable mt
+WHERE  mt.customer_id = :'customerId'
+AND    mt.date_of_purchase < :'dateOfPurchase'
+;
+`;
+
+export class MissingValueError extends Error {
+    key: string;
+    query: string;
+
+    error = 'MissingValueError';
+
+    constructor(key: string, query?: string) {
+        super(`Missing value for key \`${key}\``);
+
+        this.key = key;
+        this.query = query;
+    }
+}
+
+export interface InputParameters {
+    customerId: any;
+    dateOfPurchase: any;
+}
+
+export default function generateQuery(
+    parameters: Readonly<InputParameters>
+): PgQuery {
+    const values: any[] = [];
```

Comment:
Will you grab the name of the query and stuff it into the `name` key of the returned `PgQuery`? I particular want it for tests, to ease checking which query was passed to `client.query`. It could be something like:

```typescript
    const nameMatch = rawQuery.match(/^--\s*(?<name>\w+)/);
    const name = nameMatch.groups?.name? || null;
```

And then include that in the returned object on line 75 (as the key `name`).

Feel free to tell me that this is an unreasonable expansion of scope :)

Reply:
I like this - feels like a separate PR but one I definitely want to do!

Reply:
Implemented in #4 - https://github.com/logixboard/sureql/pull/4/commits/df9394c10a83909d599a7415c0cbabf3bfb8c43a